src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -532,8 +538,16 @@ private void OutputResultsHelper(IEnumerable<CommandInfo> results)
                         }
                         else
                         {
-                            // Write output as normal command info object.
-                            WriteObject(result);
+                            if (UseFuzzyMatching)
+                            {
+                                PSObject obj = new PSObject(result);
+                                obj.Properties.Add(new PSNoteProperty("Score", _commandScores.First(x => x.Command == result).Score));
```

Comment:
This will do a search in all items in `_commandScores` every time when writing out an object. It will likely take more time for the suggestion evaluation.

```suggestion
                                obj.Properties.Add(new PSNoteProperty("Score", _commandScores[count].Score);
```